### ISSUE TITLE
POD settings.jsonc parsing fails silently

### ISSUE THREAD

**Issue by jkrasting** [NOAA-GFDL/MDTF-diagnostics#179]
If there is an error in a POD settings.jsonc file, the framework dies with a rather unintuitive error messge:

```
  File "MDTF-diagnostics/src/diagnostic.py", line 493, in from_config
    return cls.from_struct(pod_name, config.pod_data[pod_name])
KeyError: 'some_failed_pod'
```

If the parsing for a particular POD fails, the POD is appended to a `bad_pods` list but no warning is given.  This makes troubleshooting difficult

https://github.com/NOAA-GFDL/MDTF-diagnostics/blob/70622b2fb9395de3a8ac3bc92c55c955e3d489b2/src/mdtf_info.py#L48-L54

@wrongkindofdoctor, @tsjackson-noaa  -- any thoughts on this?
